auth/server/src/main/java/com/seagates3/dao/ldap/LdapStore.java

```diff
@@ -15,99 +15,124 @@
-      method.invoke(instance, obj);
+      Method method = storeClass.getMethod(METHOD_SAVE, Object.class);
+      method.invoke(instance, entry.getValue());
     }
     catch (ClassNotFoundException e) {
+      LOGGER.error("Failed to save - " + prefix);
+      throw new DataAccessException("failed to save.\n" + e);
     }
     catch (NoSuchMethodException | SecurityException e) {
+      LOGGER.error("Exception while calling save method");
+      throw new DataAccessException("failed to call save method\n" + e);
     }
     catch (IllegalAccessException | IllegalArgumentException |
            InvocationTargetException | InstantiationException e) {
+      LOGGER.error("Failed to save - " + prefix);
+      throw new DataAccessException("failed to save- " + prefix + e);
     }
   }
 
-  @Override public Object find(String strToFind, Object obj,
+  @Override public Object find(String strToFind, Object obj, Object obj2,
                                String prefix) throws DataAccessException {
-
-    String methodName = "find";
-    String className = "com.seagates3.dao.ldap." + prefix + "LdapStore";
-    LOGGER.debug("calling method - " + methodName + " of class - " + className);
+    String className = CLASS_PACKAGE + prefix + CLASS_SUFFIX;
+    LOGGER.debug("calling method - " + METHOD_FIND + " of class - " +
+                 className);
     try {
       Class < ? > storeClass = Class.forName(className);
       Object instance = storeClass.newInstance();
-      Method method = storeClass.getMethod(methodName, Object.class);
-      Object returnObj = method.invoke(instance, obj);
+      Method method =
+          storeClass.getMethod(METHOD_FIND, Object.class, Object.class);
+      Object returnObj = method.invoke(instance, obj, obj2);
       return returnObj;
     }
     catch (ClassNotFoundException e) {
-      LOGGER.error("ClassNotFoundException" + e);
+      LOGGER.error("Exception occurred in find " + e);
+      throw new DataAccessException("failed to find - " + prefix + e);
     }
     catch (NoSuchMethodException | SecurityException e) {
-      LOGGER.error("NoSuchMethodException" + e);
+      LOGGER.error("Exception occurred " + e);
+      throw new DataAccessException("failed to find " + prefix + e);
     }
     catch (IllegalAccessException | IllegalArgumentException |
            InvocationTargetException | InstantiationException e) {
-      LOGGER.error("IllegalArgumentException " + e);
+      LOGGER.error("Exception found " + e);
+      throw new DataAccessException("failed to find " + prefix + e);
     }
-    return null;
   }
 
   @Override public List findAll(String strToFind, Object obj,
                                 String prefix) throws DataAccessException {
-    String methodName = "findAll";
-    String className = "com.seagates3.dao.ldap." + prefix + "LdapStore";
-    LOGGER.debug("calling method - " + methodName + " of class - " + className);
+    String className = CLASS_PACKAGE + prefix + CLASS_SUFFIX;
```

Comment:
className can be const and defined at one place

Reply:
prefix is coming as parameter here.
So declared it in every method